modules/data-entry/src/main/java/ca/sickkids/ccm/lfs/DeleteServlet.java

```diff
@@ -0,0 +1,364 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package ca.sickkids.ccm.lfs;
+
+import java.io.IOException;
+import java.io.Writer;
+import java.util.ArrayList;
+
+import javax.jcr.AccessDeniedException;
+import javax.jcr.Node;
+import javax.jcr.PropertyIterator;
+import javax.jcr.RepositoryException;
+import javax.jcr.Session;
+import javax.json.Json;
+import javax.json.stream.JsonGenerator;
+import javax.servlet.Servlet;
+import javax.servlet.ServletException;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.SlingHttpServletResponse;
+import org.apache.sling.api.resource.ResourceResolver;
+import org.apache.sling.api.servlets.SlingAllMethodsServlet;
+import org.apache.sling.servlets.annotations.SlingServletResourceTypes;
+import org.osgi.service.component.annotations.Component;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A servlet that tries to delete a given node and returns an explination if deletion is not possible.
+ * <p>
+ * This servlet supports the following parameters:
+ * </p>
+ * <ul>
+ * <li><tt>path</tt>: a path to the item which should be deleted</li>
+ * <li><tt>recursive</tt>: whether nodes which reference the item should be deleted; defaults to false</li>
+ * </ul>
+ *
+ * @version $Id$
+ */
+
+@Component(service = { Servlet.class })
+@SlingServletResourceTypes(
+    resourceTypes = { "lfs/QuestionnairesHomepage", "lfs/FormsHomepage", "lfs/SubjectsHomepage",
+        "lfs/SubjectTypesHomepage" },
+    extensions = { "delete" },
+    methods = { "POST" })
```

Comment:
It is not RESTful to do it this way. The resource that you request should be the resource that you want to delete.

```suggestion
    resourceTypes = { "lfs/Questionnaire", "lfs/Form", "lfs/Subject", "lfs/SubjectType" },
    methods = { "DELETE" })
```